model/roles.go

```diff
@@ -168,8 +174,57 @@ func (r *Role) GetScriptPaths() map[string]string {
 
 }
 
+// GetScriptSignatures returns the SHA1 of all of the script file names and contents
+func (r *Role) GetScriptSignatures() (string, error) {
+	hasher := sha1.New()
+
+	var scripts []string
+
+	for _, f := range r.GetScriptPaths() {
+		scripts = append(scripts, f)
```

Comment:
`GetScriptPaths()` returns a map, so using a loop was the only way. Made it a bit more efficient by allocating the array ahead of time.